collections/ansible_collections/massopencloud/esi/roles/subnet/tasks/create_subnet.yaml

```diff
@@ -0,0 +1,27 @@
+- name: Check if subnet exists
+  openstack.cloud.subnets_info:
+    name: "{{ subnet_name_or_id }}"
+  register: subnets
+
+- name: Set create_subnet_result to found subnet
+  when: subnets.subnets
+  ansible.builtin.set_fact:
+    create_subnet_result: "{{ subnets.subnets[0].id }}"
+
+- name: Create a new subnet
+  when: not subnets.subnets
+  block:
+  - name: Create subnet  # noqa:no-changed-when
```

Comment:
Do we need the `noqa` annotations here? These are necessary when using `command` tasks, but we're not doing that here.